From d3a5fce55e5668f7dd758c3aadd7913adf4594c3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20P=C5=91cze?=
 <barnabas.pocze@ideasonboard.com>
Date: Mon, 20 Jul 2026 13:21:26 +0200
Subject: [PATCH 43/68] ipa: softisp: Store exposure as `uint32_t`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Every other IPA module does that, so switch to `uint32_t` here as well.

Signed-off-by: Barnabás Pőcze <barnabas.pocze@ideasonboard.com>
Reviewed-by: Milan Zamazal <mzamazal@redhat.com>
Reviewed-by: Stefan Klug <stefan.klug@ideasonboard.com>
(cherry picked from commit 459e14449369e0f3552ef6b7e19623ec7a38f992)
---
 src/ipa/simple/algorithms/agc.cpp |  6 +++---
 src/ipa/simple/ipa_context.h      | 10 +++++-----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/ipa/simple/algorithms/agc.cpp b/src/ipa/simple/algorithms/agc.cpp
index e9bcb2c03..93505f766 100644
--- a/src/ipa/simple/algorithms/agc.cpp
+++ b/src/ipa/simple/algorithms/agc.cpp
@@ -67,7 +67,7 @@ Agc::Agc()
 
 void Agc::updateExposure(IPAContext &context, IPAFrameContext &frameContext, double exposureMSV)
 {
-	int32_t exposure = frameContext.sensor.exposure;
+	uint32_t exposure = frameContext.sensor.exposure;
 	double again = frameContext.sensor.gain;
 
 	double error = kExposureOptimal - exposureMSV;
@@ -87,7 +87,7 @@ void Agc::updateExposure(IPAContext &context, IPAFrameContext &frameContext, dou
 	if (factor > 1.0f) {
 		/* Scene too dark: increase exposure first, then gain. */
 		if (exposure < context.configuration.agc.exposureMax) {
-			int32_t next = static_cast<int32_t>(exposure * factor);
+			uint32_t next = exposure * factor;
 			exposure = std::max(next, exposure + 1);
 		} else {
 			double next = again * factor;
@@ -105,7 +105,7 @@ void Agc::updateExposure(IPAContext &context, IPAFrameContext &frameContext, dou
 			else
 				again = next;
 		} else {
-			int32_t next = static_cast<int32_t>(exposure * factor);
+			uint32_t next = exposure * factor;
 			exposure = std::min(next, exposure - 1);
 		}
 	}
diff --git a/src/ipa/simple/ipa_context.h b/src/ipa/simple/ipa_context.h
index 71dc22e89..910947b81 100644
--- a/src/ipa/simple/ipa_context.h
+++ b/src/ipa/simple/ipa_context.h
@@ -28,7 +28,7 @@ namespace ipa::soft {
 
 struct IPASessionConfiguration {
 	struct {
-		int32_t exposureMin, exposureMax;
+		uint32_t exposureMin, exposureMax;
 		double againMin, againMax, again10, againMinStep;
 		utils::Duration lineDuration;
 	} agc;
@@ -42,14 +42,14 @@ struct IPAActiveState {
 	ipa::ccm::ActiveState ccm;
 
 	struct {
-		int32_t exposure;
+		uint32_t exposure;
 		double again;
 		bool valid;
 	} agc;
 
 	struct {
 		uint8_t level;
-		int32_t lastExposure;
+		uint32_t lastExposure;
 		double lastGain;
 	} blc;
 
@@ -68,12 +68,12 @@ struct IPAFrameContext : public FrameContext {
 	ipa::ccm::FrameContext ccm;
 
 	struct {
-		int32_t exposure;
+		uint32_t exposure;
 		double gain;
 	} agc;
 
 	struct {
-		int32_t exposure;
+		uint32_t exposure;
 		double gain;
 	} sensor;
 
